workflow/util/util.go

```diff
@@ -473,8 +473,8 @@ func AddParamToGlobalScope(wf *wfv1.Workflow, log *log.Entry, param wfv1.Paramet
 		wf.Status.Outputs.Parameters = append(wf.Status.Outputs.Parameters, gParam)
 		wfUpdated = true
 	} else {
-		prevVal := *wf.Status.Outputs.Parameters[index].Value
-		if prevVal != *param.Value {
+		prevVal := wf.Status.Outputs.Parameters[index].Value
```

Comment:
@markterm I added this nil check to fix a test that just started failing. Do you think it could be a bug?

Reply:
@markterm bump!

Reply:
@alexec I'm sorry, I'm afraid I don't have bandwidth to look at this at the moment.